lib/FusionInventory/Agent/Task/Inventory/Generic/Rudder.pm

```diff
@@ -42,19 +44,65 @@ sub doInventory {
     # Get agent capabilities
     my @agentCapabilities = _listAgentCapabilities();
 
+    my $customProperties = _getCustomProperties(logger => $logger);
+
     my $rudder = {
         HOSTNAME => $hostname,
         UUID => $Uuid,
         AGENT => \@agents,
         SERVER_ROLES => { SERVER_ROLE => \@serverRoles },
         AGENT_CAPABILITIES => { AGENT_CAPABILITY => \@agentCapabilities },
+        CUSTOM_PROPERTIES => $customProperties,
     };
 
     $inventory->addEntry(
         section => 'RUDDER', entry => $rudder
     );
 }
 
+sub _getCustomProperties {
+    my (%params) = @_;
+    my $logger   = $params{logger};
+
+    my $custom_properties_dir = ($OSNAME eq 'MSWin32') ? 'C:\Program Files\Rudder\local\hooks.d' : '/var/rudder/local/hooks.d';
+    my $custom_properties;
+    if (-d "$custom_properties_dir") {
+        my @custom_propertes_list = ();
+        opendir(DIR, $custom_properties_dir); # or die $!;
+        # List each file in the custom_properties directory, each files being a script
+        my @ordered_script_list = sort readdir(DIR);
+        while (my $file = shift @ordered_script_list) {
+            my $script_file = $custom_properties_dir . "/" . $file;
+            if (-f $script_file) {
+                next if ($file =~ m/^\./);
+                # Ignore non-executable file, or folders
+                next unless -x $script_file;
+                my $properties = qx($script_file);
```

Comment:
I know you want here to not use runCommand() API as it won't fail if script fails as you asked us about that on IRC. But it is important then to at least add a debug2 level logging just before as is doing runCommand() so we can debug easily something is started from here and what.
Another point, does this scripts need to be run as root user ? This can look like a security hole if somebody other than root gains the privilege to write a script there.

Reply:
I can add the logger; however i'm not sure which user could be used to run the scripts, except root (or current admin user).
Plus, if someone can access as root to /var/rudder, we have other issues on our hands. @peckpeck do you have any advices on this ?

Reply:
today it is run by root, this means we have to make sure that this directory is not writable by anyone except root.
For users not having rudder, /var is writable only by root, so this is not a security problem for them

Reply:
Okay, then I'll be okay if you add a simple check to see if scripts is owned and executable by root. You should then log a warning and not start the script if it is not owned and runnable by admin.

More we add safe controls, more we can be sure nothing goes wrong for users.

Is your remarks also valid for win32 platform ? This is the most sensible...

Reply:
yes, c:\Program Files\Rudder is also owned and readable only by Administrator

Reply:
By root or by the current user

Reply:
if it is not runable, it cannot be run by the command, so the check is not interesting.
If you want a better check, you can check that when you are root, it is not writable by anyone else.